spec/models/in_person_enrollment_spec.rb

```diff
@@ -491,4 +494,36 @@
       expect(failed_enrollment_without_notification.eligible_for_notification?).to eq(false)
     end
   end
+
+  describe 'enhanced_ipp?' do
+    before do
+      allow(IdentityConfig.store).to receive(:usps_eipp_sponsor_id).and_return(usps_eipp_sponsor_id)
+    end
+
+    context 'when the enrollment sponsor ID is equal to the EIPP sponsor ID' do
+      it 'returns true' do
+        user = create(:user)
+        profile = create(:profile, gpo_verification_pending_at: 1.day.ago, user: user)
+        enrollment = create(
+          :in_person_enrollment, user: user, profile: profile,
+                                 sponsor_id: usps_eipp_sponsor_id
+        )
+
+        expect(enrollment.enhanced_ipp?).to be true
+      end
+    end
+
+    context 'when the enrollment sponsor ID does not equal the EIPP sponsor ID' do
+      it 'returns false' do
+        user = create(:user)
+        profile = create(:profile, gpo_verification_pending_at: 1.day.ago, user: user)
+        enrollment = create(
+          :in_person_enrollment, user: user, profile: profile,
+                                 sponsor_id: usps_ipp_sponsor_id
+        )
+
+        expect(enrollment.enhanced_ipp?).to be false
+      end
+    end
+  end
```

Comment:
Playing off what @eileen-nava was saying... what do you think about doing something like this?

```suggestion
  describe 'enhanced_ipp?' do
    let(:user) { create(:user) }
    let(:profile) do
      create(
        :profile, gpo_verification_pending_at: 1.day.ago,
                  user: user
      )
    end
    let(:enrollment) do
      create(
        :in_person_enrollment, user: user, profile: profile,
                               sponsor_id: sponsor_id
      )
    end

    before do
      allow(IdentityConfig.store).to receive(
        :usps_eipp_sponsor_id
      ).and_return(usps_eipp_sponsor_id)
    end

    context 'when the enrollment sponsor ID is equal to the EIPP sponsor ID' do
      let(:sponsor_id) { usps_eipp_sponsor_id }

      it 'returns true' do
        expect(enrollment.enhanced_ipp?).to be true
      end
    end

    context 'when the enrollment sponsor ID does not equal the EIPP sponsor ID' do
      let(:sponsor_id) { usps_ipp_sponsor_id }

      it 'returns false' do
        expect(enrollment.enhanced_ipp?).to be false
      end
    end
  end
```

Reply:
Alternatively you could swap the let defining sponsor_id with the enrollment let and only change the sponsor_id in the create if you think that is easier to follow.

Reply:
Actually I just figured out one other thing that would really make this nice. Factorybot is smart enough to know to create a user and profile when the enrollment is created so no need to create a user and profile. I would update `let(:sponsor_id)` with this

```ruby
      let(:enrollment) do
        create(:in_person_enrollment, sponsor_id: usps_ipp_sponsor_id)
      end
```

Then you can remove the `user` and `profile` lets so it would become this

```ruby
  describe 'enhanced_ipp?' do
    before do
      allow(IdentityConfig.store).to receive(
        :usps_eipp_sponsor_id,
      ).and_return(usps_eipp_sponsor_id)
    end

    context 'when the enrollment sponsor ID is equal to the EIPP sponsor ID' do
      let(:enrollment) do
        create(:in_person_enrollment, sponsor_id: usps_eipp_sponsor_id)
      end

      it 'returns true' do
        expect(enrollment.enhanced_ipp?).to be true
      end
    end

    context 'when the enrollment sponsor ID does not equal the EIPP sponsor ID' do
      let(:enrollment) do
        create(:in_person_enrollment, sponsor_id: usps_ipp_sponsor_id)
      end

      it 'returns false' do
        expect(enrollment.enhanced_ipp?).to be false
      end
    end
  end
```

Reply:
@shanechesnutt-ft  I used this block of code and just added in traits to create the enrollment.  Thank you